octopus_sensing/devices/skinos_streaming.py

```diff
@@ -0,0 +1,264 @@
+# This file is part of Octopus Sensing <https://octopus-sensing.nastaran-saffar.me/>
+# Copyright © Nastaran Saffaryazdi 2020
+#
+# Octopus Sensing is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+#  either version 3 of the License, or (at your option) any later version.
+#
+# Octopus Sensing is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with Octopus Sensing.
+# If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import platform
+import threading
+import datetime
+import csv
+import math
+import struct
+import serial
+from typing import List, Optional
+import time
+import socket
+
+from octopus_sensing.devices.monitored_device import MonitoredDevice
+from octopus_sensing.common.message_creators import MessageType
+from octopus_sensing.common.message import Message
+from octopus_sensing.devices.common import SavingModeEnum
+
+HOST = '127.0.0.1'
+PORT = 50007
+
+class SkinosStreaming(MonitoredDevice):
+
+    def __init__(self,
+                 sampling_rate: int=128,
+                 saving_mode: int=SavingModeEnum.CONTINIOUS_SAVING_MODE,
+                 **kwargs):
+        super().__init__(**kwargs)
+
+        self._saving_mode = saving_mode
+        self._stream_data: List[float] = []
+        self._sampling_rate = sampling_rate
+        self._inintialize_connection()
```

Comment:
It's a good idea to move the `initialization` and also `socket` creation to the `_run` method. The `_run` method will be called after the creation of the child process, so the connection to the device and the socket will happen inside the child process and not the main process.

It can become problematic on some platforms. In your code, the handler of both connections is in the main process' memory, and when the fork happens, the child process receives a copy of the handler. Sometimes, this copying might corrupt the connection. (On Windows specifically, the child process will never receive the copy.)

By moving these to the `_run` method, these will be created inside the child process, and that problem won't occur. (Also, it will be more isolated.)